### PR TITLE
changing tuning match queries to return iterators

### PR DESCRIPTION
**Pull request by ericbuckley**
## Description
[Provide a brief description of the changes implemented in this pull request.]

## Related Issues
refs #456 

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: src/recordlinker/database/mpi_service.py [hunk 1]
```diff
@@ -640,48 +626,43 @@ def generate_non_match_tuning_samples(
...
+    random_ids: list[int] = [
+        row[0]
+        for row in session.query(models.Patient.id)
+        .order_by(get_random_function(session.get_bind().dialect))
+        .limit(sample_size)
+        .all()
+    ]
+    query: expression.Select = (
+        select(models.Patient.id, models.Patient.person_id, models.Patient.data)
+        .where(models.Patient.id.in_(bindparam("ids")))
```
Comment (m-goggins) [inline src/recordlinker/database/mpi_service.py:638]:
```suggestion
       .where(models.Patient.id.in_(bindparam("ids", expanding=True)))
```
I _think_ you need to add `expanding=True` or else it might treat the whole list of ids as a single entity rather than checking if the Patient.id is in the list. See documentation [here](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.bindparam.params.expanding)

Comment (ericbuckley) [reply src/recordlinker/database/mpi_service.py:638]: I believe its working, but yeah the documentation def says its required, so I'll merge that in.  Great find!
